core/unix/coredump.c

```diff
@@ -484,11 +484,17 @@ os_dump_core_internal(dcontext_t *dcontext)
     // of LOAD program header (number of sections minus one since shstrtab does not
```

Comment:
See my readelf on hello where the section header table is *not* page-aligned:
```
  Start of section headers:          14584 (bytes into file)
```

That makes me think this padding is not necessary.  Is there some other alignment after the headers that is the real culprit?  It sure looks like we do *not* need to align the sec headers.

Reply:
I've added a pointer to https://man7.org/linux/man-pages/man5/elf.5.html, and spelled out the missing padding in the core file to align loadable segments to page size.

Reply:
We need to align loadable segments, for example:
```
Section Headers:
  [Nr] Name              Type             Address           Offset
       Size              EntSize          Flags  Link  Info  Align
  [ 0]                   PROGBITS         00007f0018e00000  00000e00
       0000000000040000  0000000000000000 WAM       0     0     8
```
The offset should be a multiple of the page size, instead of 0xe00. The offset applies to both the address and the offset in the file. 

I missed the file part in previous implementations.